src/toolbox/logging_module/logging_module_test.py

```diff
@@ -0,0 +1,148 @@
+        if not os.path.exists('./logs'):
+            os.makedirs('./logs')
+
+        # Get the file paths for testing and production.
+        if self.testing:
+            path = f"./logs/testing_{bot_name}_{datetime.today().strftime('%Y-%m-%d_%H-%M')}.log"
+        else:
+            path = f"./logs/{bot_name}_{datetime.today().strftime('%Y-%m-%d')}.log"
+        return path
+    def log(self, message, level=logging.INFO):
+        """
+        Loggea el mensaje en memoria
+        """
+        self.logger.log(level, message)
+    def flush(self):
+        """
+        Escribe los mensajes en memoria en el archivo de log
+        """
+        for handler in self.logger.handlers:
+            handler.flush()
+            handler.close()
+
+class MyNormalLogger():
+    """
+    Logger que escribe los mensajes en el archivo de log a medida que se van recibiendo.
+    """
+    def __init__(self, bot_name: str, testing : bool = True) -> None:
+        """Inicializa el logger
+
+        Parameters
+        ----------
+        bot_name: str
+            Nombre del bot
+
+        testing : bool
+            Referencia el ambiente
+        """
+        self.testing = testing
+        self.logger = logging.getLogger(__name__)
+        self.path_to_log_file = self.get_path_to_log_file(bot_name)
+        file_handler = logging.FileHandler(self.path_to_log_file)
+        formatter = logging.Formatter("%(asctime)s :: %(funcName)s :: %(lineno)d :: %(message)s")
+        file_handler.setFormatter(formatter)
+        self.logger.addHandler(file_handler)
+        self.logger.setLevel(logging.DEBUG)
+    def get_path_to_log_file(self, bot_name: str) -> str:
+        """
+        Devuelve el path al archivo de logging
+        """
+
+        # Check if the logs folder exists, and if not, create it
+        if not os.path.exists('./logs'):
+            os.makedirs('./logs')
+
+        # Get the file paths for testing and production.
+        if self.testing:
+            path = f"./logs/testing_{bot_name}_{datetime.today().strftime('%Y-%m-%d_%H-%M')}.log"
+        else:
+            path = f"./logs/{bot_name}_{datetime.today().strftime('%Y-%m-%d')}.log"
+        return path
+    def log(self, message, level=logging.INFO):
+        """
+        Loggea el mensaje en el archivo de log
+        """
+        self.logger.log(level, message)
+
+def test_time_difference(n_logs: int, cap: int, message_length: int = 100):
+    """
+    Testea la diferencia en velocidad entre el BatchLogger y el NormalLogger.
+    
+    Parámetros
+    ----------
+    n_logs: int
+        Número de mensajes para loggear.
+    def test_time_difference(n_logs: int, cap: int, message_length: int = 100):
+    cap: int
+        Capacidad del BatchLogger.
+    message_length: int
+        Longitud del mensaje a Loggear.
+    """
+    tic = time.time()
+    logger = MyBatchLogger('batch', capacity=cap)
+    for i in range(n_logs):
+        logger.log('L'*message_length, level=logging.DEBUG)
```

Comment:
No estaría bueno agregar en el momento que llega el log  para poder analizar y ver la velocidad? Saque la idea de como llegan los datos de mercado: `{'type': 'Md', 'timestamp': 1693320605687, 'instrumentId': {'marketId': 'ROFX', 'symbol': 'MERV - XMEV - AL41 - CI'}, 'marketData': {'OF': [{'price': 23978.5, 'size': 550}]}}`. Digo de agregar un timestamp, sumado a la hora que le agrega el logging.

Reply:
No entendi, estas hablando de agregar a los logs un timestamp?
O de agregar otro timer? O ninguna de las dos?

Reply:
Agregar un timestamp como en el ejemplo que te pase (no necesariamente el mismo formato) para poder ver cuando realmente tuve ese log y cuando lo escribí, que con el método batch no necesariamente son iguales